src/MerchantCenter/MerchantCenterService.php

```diff
@@ -227,4 +277,75 @@ protected function saved_target_audience(): bool {
 		$empty_selection = 'selected' === $audience['location'] && empty( $audience['countries'] );
 		return ! $empty_selection;
 	}
+
+	/**
+	 * Checks if we should add an issue when the contact information is not setup.
+	 *
+	 * @since x.x.x
+	 *
+	 * @param array    $issues             The current array of custom issues
+	 * @param DateTime $cache_created_time The time of the cache/issues generation.
+	 *
+	 * @return array
+	 */
+	protected function maybe_add_contact_info_issue( array $issues, DateTime $cache_created_time ): array {
+		if ( $this->is_setup_complete() && ! $this->is_contact_information_setup() ) {
+			$issues[] = [
+				'product_id' => 0,
+				'product'    => 'All products',
+				'code'       => 'missing_contact_information',
+				'issue'      => __( 'No contact information.', 'google-listings-and-ads' ),
+				'action'     => __( 'Add store contact information', 'google-listings-and-ads' ),
+				'action_url' => $this->get_contact_information_setup_url(),
+				'created_at' => $cache_created_time->format( 'Y-m-d H:i:s' ),
+				'type'       => MerchantStatuses::TYPE_ACCOUNT,
+				'severity'   => 'error',
+				'source'     => 'filter',
+			];
+		}
+
+		return $issues;
+	}
+
+	/**
+	 * Check if the Merchant Center contact information has been setup already.
+	 * The check results are stored in a transient to prevent multiple requests to the API.
+	 *
+	 * @since x.x.x
+	 *
+	 * @return boolean
+	 */
+	protected function is_mc_contact_information_setup(): bool {
+		/** @var TransientsInterface $transients */
+		$transients = $this->container->get( TransientsInterface::class );
+		$is_setup   = $transients->get( TransientsInterface::MC_CONTACT_INFO );
```

Comment:
Instead of setting a new transient wouldn't it make sense to update the `OptionsInterface::CONTACT_INFO_SETUP` option here in case both phone number and address are set up?

This method seems to be used only if that option doesn't exist. And that option is also updated based on the result of this method when called in `MerchantCenterService::is_contact_information_setup`. So it probably makes sense to just update the option and avoid a transient in between.

Reply:
In my original code the transient was more relevant, so yes you are right that it's not as relevant to rely on both an option and a transient. Although the option is only used if it's value is true, otherwise it will once again do the check to see if the Merchant Center is setup. Although there are still the following checks in place which prevent it from being called repeatedly.

The notification will be checked on every request, although it has a check to see if the notification has already been added it won't be added again (which ensures the contact information will only be checked the first time round).

For the issue being added to the table, we already have another transient for the product statuses, since we hook into when this data is refreshed (once an hour), we will only do the check every time the transient expires.

In theory though both those two checks can happen on the same request (which is rare but possible), so in that case it will send two requests to check the contact information. For this reason I have left the transient in place so it wouldn't repeat. Since it's a rare circumstance do you think we should remove it to simplify the code? We can also decide to move it further down the line when there will be less people using the plugin without having gone through the contact information step.

Reply:
Hmm. I think i's okay to do the API call whenever the MC status cache is deleted because we need to have the most updated state in there. And it's only temporary because as you said once they set a phone and address we just check and return the value in the options. 

Should we care if they set a phone number now but then delete it once we set the option value to `true`? Because I think we wouldn't be able to tell if they do that.

Reply:
Alright, it makes sense and I'll look into removing the transient then.

As far as I understand it's our main concern to prompt them to enter the information if they haven't done so yet. You can't delete a phone number through the MC dashboard, but if they do manage to clear it somehow then the consequence will be "If you do not add your contact information, your listings may not appear on Google". I don't think we need to actively check that after we've already got them to enter it once.